library/std/src/thread/mod.rs

```diff
@@ -684,9 +684,12 @@ thread_local! {
 
 /// Sets the thread handle for the current thread.
 ///
-/// Panics if the handle has been set already or when called from a TLS destructor.
+/// Aborts if the handle has been set already to reduce code size.
 pub(crate) fn set_current(thread: Thread) {
-    CURRENT.with(|current| current.set(thread).unwrap());
+    CURRENT.with(|current| {
+        rtassert!(current.get().is_none());
+        current.set(thread).unwrap();
```

Comment:
could you `match`+`rtabort` instead to avoid still having an (optimized out) `unwrap`?